app/src/gui/components/record/formButton.tsx

```diff
@@ -76,8 +80,34 @@ export default function FormButtonGroup(props: any) {
     views,
     ui_specification,
     mq_above_md,
+    relationship,
     buttonRef,
   } = props;
+
+  const [labels, setLabels] = useState({
+    continue: 'Publish and continue editing',
+    close: 'Publish and close record',
+    new: 'Publish and new record',
+  });
+
+  useEffect(() => {
+    const get = async () => {
+      if (relationship.parent) {
+        const parentName = await getRecordType(
+          project_id,
+          relationship.parent.record_id
+        );
+
+        setLabels({
+          ...labels,
+          close: `Save and return to ${parentName}`,
+          new: `Publish and new ${record_type}`,
+        });
+      }
+    };
+    get();
+  }, []);
```

Comment:
might want to put project_id and relationship into the dependency array - seems weird that the relationship can't just pass through directly the record types rather than doing this async fetch thing - but is what it is

Reply:
done